target/snitch_cluster/Makefile

```diff
@@ -218,12 +232,30 @@ test/bootrom.elf test/bootrom.dump test/bootrom.bin: test/bootrom.S test/bootrom
 	riscv64-unknown-elf-objcopy -j .text -O binary test/bootrom.elf test/bootrom.bin
 
 
-$(SNAX_GEN):
-	${WRAPPER_GEN} --cfg_path="$(ROOT)/target/snitch_cluster/$(CFG_OVERRIDE)" \
+SNAX_TEMPLATE_SOURCES = $(wildcard $(SNAX_TPL_PATH)*.tpl)
+CFG_FILE = $(ROOT)/target/snitch_cluster/$(CFG_OVERRIDE)
+SNAX_SCALA_SOURCES = \
+	$(wildcard ${CHISEL_PATH}/src/main/scala/snax/streamer/*.scala) \
+	$(wildcard ${CHISEL_PATH}/src/main/scala/snax/csr_manager/*.scala)
+
+# Pick the first file
+FIRST_SNAX_GEN := $(firstword $(SNAX_GEN))
+# Pick all the files but the first file, if there are more than one
+OTHER_SNAX_GEN := $(filter-out $(FIRST_SNAX),$(SNAX_GEN))
+
+# Force a serialized build in parallel make through a circular dependency
+# Implemented as documented in:
+# https://www.gnu.org/software/automake/manual/html_node/Multiple-Outputs.html
+# Please note the limitations of this method over there
+$(FIRST_SNAX_GEN) : $(SNAX_TEMPLATE_SOURCES) $(CFG_FILE) $(SNAX_SCALA_SOURCES)
+	${WRAPPER_GEN} --cfg_path="$(CFG_FILE)" \
 	--tpl_path="${SNAX_TPL_PATH}" \
 	--chisel_path="${CHISEL_PATH}" \
 	--gen_path="${GENERATED_DIR}/"
 
+# This is a fake job that forces parallel make to wait for the first thing to be generated
+$(OTHER_SNAX_GEN): $(FIRST_SNAX_GEN)
```

Comment:
Such work around. hehehe 😆

Reply:
It -  unfortunately -  is the recommended method as per the GNU docs: https://www.gnu.org/software/automake/manual/html_node/Multiple-Outputs.html